triage/labels.rst

```diff
@@ -10,405 +10,136 @@ Many labels are shared for both use cases, while some are dedicated
 only to one. Below is a possibly inexhaustive list, but it should get
 you going. For a full list, see `here <https://github.com/python/cpython/issues/labels>`_.
 
-General purpose labels
-======================
 
-:gh-label:`type-behavior`
-    Used for issues/PRs that address unintentional behavior, but do not
-    pose significant security concerns. Generally, bugfixes will be attached
-    to a specific issue where the unintended behavior was first reported.
+.. _general-purpose-labels:
+.. _Type:
+
+Type labels
+===========
+
+These labels are used to specify the type of issue:
 
-:gh-label:`type-documentation`
-    Used for issues/PRs that exclusively involve changes to
-    the documentation. Documentation includes ``*.rst`` files, docstrings,
-    and code comments.
+* :gh-label:`type-bug`: for unexpected behaviors, bugs, or exceptions
+  (not hard crashes).
+* :gh-label:`type-crash`: for hard crashes of the interpreter, possibly with a
+  core dump.
+* :gh-label:`type-feature`: for feature requests or enhancements.
+  The `Ideas Discourse category`_ can be used to discuss enhancements
+  before filing an issue.
+* :gh-label:`type-security`: for security issues.
+  See also `Reporting security issues in Python`_.
 
-:gh-label:`type-enhancement`
-    Used for issues/PRs that provide additional functionality
-    or capabilities beyond the existing specifications.
 
-:gh-label:`type-performance`
-    Used for issues/PRs that provide performance optimizations.
+.. _Component:
 
-:gh-label:`type-security`
-    Used for issues/PRs that involve critical security issues. Less severe
-    security concerns can instead use the type-bugfix label.
+Component labels
+================
 
-:gh-label:`type-tests`
-    Used for issues/PRs that exclusively involve changes to the tests.
+These labels are mostly used to specify which :ref:`part of the codebase
+<directory-structure>` is affected by the issue/PR:
 
-:gh-label:`OS-Mac` / :gh-label:`OS-Windows`
-    Used for issues/PRs involving changes which only have an effect upon
-    a specific operating system.
+* :gh-label:`stdlib`: for standard library modules in the :cpy-file:`Lib`
+  directory (written in Python).
+* :gh-label:`extension-modules`: for standard library modules in the
+  :cpy-file:`Modules` directory (written in C).
+* :gh-label:`interpreter-core`: for changes related to the interpreter core in
+  the :cpy-file:`Objects`, :cpy-file:`Python`, :cpy-file:`Grammar`,
+  and :cpy-file:`Parser` dirs (written mostly in C).
+* :gh-label:`docs`: for documentation in the :cpy-file:`Doc` directory
+  (written in :ref:`reStructuredText <markup>`), docstrings, and code comments.
+* :gh-label:`tests`: for tests in the :cpy-file:`Lib/test` directory
+  (written in Python) and other changes related to tests, ``unittest``,
+  or ``doctests``.
```

Comment:
```suggestion
  (written in Python) and other changes related to tests, :mod:`unittest`,
  or :mod:`doctests`.
```

* Cross-reference and format the relevant modules
* Be consistent about using either the module name literal vs. referring to the tests themselves in prose